### ISSUE TITLE
Search misses item - race condition?

### ISSUE THREAD

**Issue by ile** [arangodb/arangodb#2072]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [x] 3.0.9

On this operating system:
  - [x] Ubuntu .deb

I'm issuing AQL via:
- [x] this Driver: arangojs

I've run `db._explain("<my aql query>")` and it didn't shed more light on this.
The AQL query in question is: [testcase.tar.gz](https://github.com/arangodb/arangodb/files/488439/testcase.tar.gz)

The issue can be reproduced using this dataset: [testcase.tar.gz](https://github.com/arangodb/arangodb/files/488439/testcase.tar.gz)

Please provide a way to create the dataset to run the above query on; either by a gist with an arangodump, or `db.collection.save({my: "values"}) statements. If it can be reproduced with one of the ArangoDB example datasets, it's a plus.

These are the steps to reproduce:
1) create the dataset
2) create the index:  db.testc.ensureIndex({ type: "fulltext", fields: [ "name" ], minLength: 2 });
3) install node
4) node index.js

The following problem occurs: The search misses one item occasionally, the one that is being written to.

Instead I would be expecting:  I should be getting all the three items.

```
$ node index.js 
[ '13483222', '13483251', 'abc' ]
[ '13483222', '13483251', 'abc' ]
[ '13483222', '13483251', 'abc' ]
[ '13483222', '13483251', 'abc' ]
[ '13483222', '13483251', 'abc' ]
[ '13483222', '13483251', 'abc' ]
[ '13483222', '13483251', 'abc' ]
[ '13483222', '13483251', 'abc' ]
[ '13483222', '13483251' ]
error
```

**Comment by jsteemann** [comment]
This was indeed a race condition. It is fixed now in 3.0.10 and devel. @ile: thanks for reporting this!

**Comment by dothebart** [comment]
3.0.10 is available for download now.
